pilz_robot_programming/src/pilz_robot_programming/commands.py

```diff
@@ -204,14 +204,19 @@ def __str__(self):
 
     def _cmd_to_request(self, robot):
         """Transforms the given command to a MotionPlanRequest."""
-        self._robot = robot
         req = MotionPlanRequest()
 
+        self._robot_reference_frame = robot._robot_commander.get_planning_frame()
+        self._active_joints = robot._robot_commander.get_group(self._planning_group).get_active_joints()
+        self._start_joint_states = robot.get_current_joint_states(planning_group=self._planning_group)
+        self._start_pose = robot.get_current_pose(target_link=self._target_link, base=self._reference_frame)
+        self._tf_listener = robot.tf_listener_
```

Comment:
- How about adding methods `get_reference_frame()` and `get_active_joints()` to the Robot class?
- How about injecting a method for the pose transformation by the robot? This would loose the coupling even more.

Reply:
done.